src/query-builder/transformer/DocumentToEntityTransformer.ts

```diff
@@ -177,8 +181,11 @@ export class DocumentToEntityTransformer {
                 }
             });
         });*/
+        const broadcastResult = new BroadcasterResult();
+        this.queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, [entity]);
+        if (broadcastResult.promises.length > 0) Promise.all(broadcastResult.promises).then(entity => entity);
```

Comment:
This is main problem in your PR. You are running async operation like `Promise.all(broadcastResult.promises).then(entity => entity);` inside sync method. I think you are lucky your tests are running successfully. 

Note: I hate one line if statemens 😄

Reply:
So I'm should run this operation inside `cursor.toArray` ?

Reply:
when run inside `cursor.toArray` worked properly
```js
return ParentCursor.prototype.toArray.call(this).then(async (results: Entity[]) => {
                    const transformer = new DocumentToEntityTransformer();
                    const entities = transformer.transformAll(results, metadata);
                    const broadcastResult = new BroadcasterResult();
                    queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, entities);
                    if (broadcastResult.promises.length > 0) await Promise.all(broadcastResult.promises);
                    return entities;
                });
```

Reply:
Yeah. Whole this `applyEntityTransformationToCursor` is really big hack of mongo 😄 

```typescript
protected applyEntityTransformationToCursor<Entity>(metadata: EntityMetadata, cursor: Cursor<Entity> | AggregationCursor<Entity>) {
        const ParentCursor = PlatformTools.load("mongodb").Cursor;
        const queryRunner = this.queryRunner;
        cursor.toArray = function (callback?: MongoCallback<Entity[]>) {
            if (callback) {
                ParentCursor.prototype.toArray.call(this, (error: MongoError, results: Entity[]): void => {
                    if (error) {
                        callback(error, results);
                        return;
                    }

                    // transform
                    const transformer = new DocumentToEntityTransformer();
                    const entities = transformer.transformAll(results, metadata);

                    // broadcast
                    const broadcastResult = new BroadcasterResult();
                    queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, entities);

                    Promise.all(broadcastResult.promises).then(() => {
                        return callback(error, entities);
                    });
                });
            } else {
                return ParentCursor.prototype.toArray.call(this).then((results: Entity[]) => {
                    // transform
                    const transformer = new DocumentToEntityTransformer();
                    const entities = transformer.transformAll(results, metadata);

                    // broadcast
                    const broadcastResult = new BroadcasterResult();
                    queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, entities);

                    return Promise.all(broadcastResult.promises);
                });
            }
        };
        cursor.next = function (callback?: MongoCallback<CursorResult>) {
            if (callback) {
                ParentCursor.prototype.next.call(this, (error: MongoError, result: CursorResult): void => {
                    if (error || !result) {
                        callback(error, result);
                        return;
                    }

                    // transform
                    const transformer = new DocumentToEntityTransformer();
                    const entities = transformer.transform(result, metadata);

                    // broadcast
                    const broadcastResult = new BroadcasterResult();
                    queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, entities);

                    Promise.all(broadcastResult.promises).then(() => {
                        return callback(error, entities);
                    });
                });
            } else {
                return ParentCursor.prototype.next.call(this).then((result: Entity) => {
                    if (!result) return result;

                    // transform
                    const transformer = new DocumentToEntityTransformer();
                    const entities = transformer.transform(result, metadata);

                    // broadcast
                    const broadcastResult = new BroadcasterResult();
                    queryRunner.broadcaster.broadcastLoadEventsForAll(broadcastResult, metadata, entities);

                    return Promise.all(broadcastResult.promises);
                });
            }
        };
    }
```